pkg/http/api.go

```diff
@@ -813,6 +824,28 @@ func (a *api) onPublish(c *routing.Context) error {
 	return nil
 }
 
+func (a *api) onPerformTransaction(c *routing.Context) error {
+	body := c.PostBody()
+
+	var requests = []state.TransactionalRequest{}
+	err := a.json.Unmarshal(body, &requests)
+	if err != nil {
+		msg := NewErrorResponse("ERR_MALFORMED_REQUEST", err.Error())
+		respondWithError(c.RequestCtx, 400, msg)
+		return nil
+	}
+
+	err = a.actor.PerformTransaction(&requests)
```

Comment:
it should not call into pkg/actors.go for making state changes as actors modify the keys for storage, Actors pkg deals with actor specific logic